--- /etc/rc.subr.orig	2014-12-18 21:07:36.000000000 +0100
+++ /etc/rc.subr	2014-12-18 21:09:44.000000000 +0100
@@ -1330,24 +1330,17 @@
 		_rc_conf_loaded=true
 	fi
 
-	# If a service name was specified, attempt to load
-	# service-specific configuration
-	if [ -n "$_name" ] ; then
-		for _d in /etc ${local_startup%*/rc.d}; do
-			if [ -f ${_d}/rc.conf.d/"$_name" ]; then
-				debug "Sourcing ${_d}/rc.conf.d/$_name"
-				. ${_d}/rc.conf.d/"$_name"
-			elif [ -d ${_d}/rc.conf.d/"$_name" ] ; then
-				local _rc
-				for _rc in ${_d}/rc.conf.d/"$_name"/* ; do
-					if [ -f "$_rc" ] ; then
-						debug "Sourcing $_rc"
-						. "$_rc"
-					fi
-				done
-			fi
-		done
-	fi
+	for _d in /etc ${local_startup%*/rc.d}; do
+		if [ -d ${_d}/rc.conf.d ] ; then
+			local _rc
+			for _rc in ${_d}/rc.conf.d/* ; do
+				if [ -f "$_rc" ] ; then
+					debug "Sourcing $_rc"
+					. "$_rc"
+				fi
+			done
+		fi
+	done
 
 	# Set defaults if defined.
 	for _var in $rcvar $rcvars; do
